test/peripheral_test.dart

```diff
@@ -0,0 +1,407 @@
+      verify(managerForPeripheral.readCharacteristicForDevice(any, any, any, '4'));
+      verify(managerForPeripheral.readCharacteristicForDevice(any, any, any, '9'));
+    });
+
+    test("should write value to characteristic", () async {
+      //given
+      CharacteristicWithValue mockedCharacteristicWithValue  = CharacteristicWithValueMock();
+      Uint8List value = Uint8List.fromList([1, 4, 9]);
+      const withResponse = false;
+
+      when(managerForPeripheral.writeCharacteristicForDevice(any, SERVICE_UUID, CHARACTERISTIC_UUID, value, withResponse, TRANSACTION_ID))
+          .thenAnswer((_) => Future.value(mockedCharacteristicWithValue));
+
+      //when
+      CharacteristicWithValue characteristicWithValue = await peripheral.writeCharacteristic(SERVICE_UUID, CHARACTERISTIC_UUID, value, withResponse, transactionId: TRANSACTION_ID);
+
+      //then
+      expect(characteristicWithValue, mockedCharacteristicWithValue);
+    });
+
+    test("should use autogenerated transactionId during writing operation", () async {
+      //given
+      var ids = ["4", "9"];
+      when(transactionIdGeneratorMock.getNextId()).thenAnswer((_) => ids.removeAt(0));
+
+      //when
+      await peripheral.writeCharacteristic('s1', 'c1', Uint8List.fromList([1]), false);
+      await peripheral.writeCharacteristic('s1', 'c1', Uint8List.fromList([1]), false);
+
+      //then
+      verify(managerForPeripheral.writeCharacteristicForDevice(any, any, any, any, any, '4'));
+      verify(managerForPeripheral.writeCharacteristicForDevice(any, any, any, any, any, '9'));
+    });
+
+    test("should return values of monitored characteristic", () {
+      //given
+      var emittedValues = [
+        CharacteristicWithValueMock(),
+        CharacteristicWithValueMock(),
+        CharacteristicWithValueMock()
+      ];
+      when(managerForPeripheral.monitorCharacteristicForDevice(any, SERVICE_UUID, CHARACTERISTIC_UUID, TRANSACTION_ID))
+          .thenAnswer((_) => Stream.fromIterable(emittedValues));
+
+      //when
+      Stream<CharacteristicWithValue> characteristicsStream =  peripheral.monitorCharacteristic(SERVICE_UUID, CHARACTERISTIC_UUID, transactionId: TRANSACTION_ID);
+
+      //then
+      expect(characteristicsStream, emitsInOrder(emittedValues));
+    });
+
+    test("should use autogenerated transactionId for monitoring operation", () {
+      //given
+      var ids = ["4", "9"];
+      when(transactionIdGeneratorMock.getNextId()).thenAnswer((_) => ids.removeAt(0));
+
+      //when
+     peripheral.monitorCharacteristic('s1', 'c1');
+     peripheral.monitorCharacteristic('s1', 'c1');
+
+      //then
+      verify(managerForPeripheral.monitorCharacteristicForDevice(any, any, any, '4'));
+      verify(managerForPeripheral.monitorCharacteristicForDevice(any, any, any, '9'));
+    });
+
+  });
+
+  group("Rssi", () {
+    test("use given transactionId", () async {
+      //given
+      const transactionId = "86791384-";
+      const rssi = 23;
```

Comment:
This value would be negative in the wild. Perhaps we should set it this way in the test in a case someone new to BLE decides to reach to them for help?